packages/@aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/logging-config.ts

```diff
@@ -0,0 +1,54 @@
+import * as logs from 'aws-cdk-lib/aws-logs';
+
+/**
+ * Represents the logging configuration for error logs.
+ * 
+ * This class defines whether logging is enabled or disabled and holds
+ * the CloudWatch log group where error logs are stored if logging is enabled.
+ */
+export abstract class LoggingConfig {
```

Comment:
To understand, is there any specific reason to choose abstract class over interface based implementation, since we are not actually defining any implementation details in this class that can be used in derived classes, is this something that we plan to add in future?
If not, I would prefer to use interface and define it as a return type instead of abstract class.

Reply:
The main reason is to avoid the use of a nested structure which is mentioned in the [Design Guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat). In this case though it probably would be okay to still use an interface because anything less than 1-2 layers deep should be fine.

Reply:
If you're talking about this it makes sense to have [abstraction class here](https://github.com/paulhcsun/aws-cdk/commit/8e9218525b606d72b2dfe55933fa1c515d26d386#diff-ffc33b42d6fe8a283d555b65510248d56a7ad07c51e8a86cdc7c39e4a39ca65bR32), i don't think this will introduce any nesting unless i'm wrong here (as I;ve seen in the case of vpc to be the return type as Interface rather than an abstract class) and end UX should remain same, its just that the binding contract will be a interface type and not a class. Let me know if it makes sense.